### ISSUE TITLE
Wrong path to config directory on multisite installation

### ISSUE THREAD

**Issue by petiar** [backdrop/backdrop-issues#666]
Hi everybody,

after installing Backdrop CMS as multisite installation I am getting this error:

`Exception: The configuration directory in settings.php is specified as 'files/config_6e725e8138933c0593de000d7d5bb4e7/active', but this directory is either empty or missing crucial files. Check that the $config_directories variable is correct in settings.php. in _backdrop_bootstrap_configuration() (line 2668 of /data/web/backdrop/core/includes/bootstrap.inc).`

while trying to access the homepage. After some digging I have found out that although my config files are successfully created in `sites/studiozavodou.sk/files`, Backdrop is trying to get them from `files/ directory` in installation root. I have proved this by copying my config files to this directory and this error disappeared.

I have tried to changed the config directory path in `settings.php` so I added the `sites/studiozavodou.sk` part at the begining of it, however, this caused another error, because it seems to that when Backdrop is trying to **update** config files, it uses correct base directory (`sites/studiozavodou.sk`), so this part got duplicated and Backdrop threw another exception, trying to update file in `sites/studiozavodou.sk/sites/studiozavodou.sk/files...`.

## Long story short (conclusion): ##
On multisite installation, Backdrop does not take the multisite base path into account while reading config files, but it does while trying to update them.

## Possible solution??? ##

On the line 2666 in `core/includes/bootstrap.inc` I have changed the code like this and now all works fine (I have also check on another non-multisite installation and it works ok):

`$directory = conf_path().'/'.$GLOBALS['config_directories']['active'];`

**Comment by quicksketch** [comment]
Hi @petiar, thanks for opening this issue. I think you're right that some funny business is happening when specifying config directories. We have this bit of code in `config_get_config_directory()`:

```php
    // If the path starts with a slash or dot, assume a normal path. If just
    // a directory name is provided, make it relative to the settings.php file.
    $first_character = substr($path, 0);
    if (!in_array($first_character, array('.', '/', '\\'))) {
      $path = conf_path() . '/' . $path;
    }
```

Your solution basically does the same thing as this code, where conf_path() is being prepended automatically to a relative path. However, your solution won't treat absolute paths correctly. I think we just need to make that line (2666) use config_get_config_directory() instead of checking the global directly.

**Comment by quicksketch** [comment]
Could you try out the patch in https://github.com/backdrop/backdrop/pull/717.patch?

**Comment by petiar** [comment]
Hi @quicksketch 

thanks for the response, your patch works alright. My solution was based just on, literary, full text search of backdrop source files for the "multisite" string, as I am not familiar with the code, so when I found the conf_path() function it looked like I discovered the holly grail of configuration settings. :-)

I am now going to close this issue as I consider it solved. Thanks for your support, hopefully we'll see this patch coming in the next backdrop release.

**Comment by quicksketch** [comment]
Excellent thanks for testing! I've merged https://github.com/backdrop/backdrop/pull/717. This will be in 1.0.2.
